src/NzbDrone.Core/IndexerSearch/ReleaseSearchService.cs

```diff
@@ -548,9 +548,10 @@ private List<DownloadDecision> Dispatch(Func<IIndexer, IEnumerable<ReleaseInfo>>
 
         private List<DownloadDecision> DeDupeDecisions(List<DownloadDecision> decisions)
         {
-            // De-dupe reports by guid so duplicate results aren't returned. Pick the one with the least rejections.
-
-            return decisions.GroupBy(d => d.RemoteEpisode.Release.Guid).Select(d => d.OrderBy(v => v.Rejections.Count()).First()).ToList();
+            // De-dupe reports by guid so duplicate results aren't returned. Pick the one with the least rejections and higher indexer priority.
+            return decisions.GroupBy(d => d.RemoteEpisode.Release.Guid)
+                .Select(d => d.OrderBy(v => v.Rejections.Count()).ThenBy(v => v.RemoteEpisode?.Release?.IndexerPriority ?? 100).First())
```

Comment:
The default priority is 25: https://github.com/Sonarr/Sonarr/blob/1a4403e0ab9ab824c97acf97174f894b3770f2ef/src/NzbDrone.Core/Indexers/IndexerDefinition.cs#L14 though worth extracting to a constant so we're not using a magic number

Reply:
Should be pretty rare that we have duplicates with the same GUID and no indexer, since that'd mostly come from pushed releases I think.